File: 50. Pow(x, n)/50. Pow(x, n).cs
public class Solution {
    public double MyPow(double x, int n) {
        if (n == 0) return 1;
        if (n == int.MinValue) {
            x = 1 / x;
            n = -(n + 1);
            return x * MyPow(x, n);
        }
        if (n < 0) {
            x = 1 / x;
            n = -n;
        }
        double result = 1;
        while (n > 0) {
            if ((n & 1) == 1) result *= x;
            x *= x;
            n >>= 1;
        }
        return result;
    }
}

File: 50. Pow(x, n)/50. Pow(x, n).rs
impl Solution {
    pub fn my_pow(x: f64, n: i32) -> f64 {
        if n == 0 {
            return 1.0;
        }
        if n == i32::MIN {
            let x = 1.0 / x;
            let n = -(n + 1);
            return x * Self::my_pow(x, n);
        }
        let (mut x, mut n) = if n < 0 {
            (1.0 / x, -n)
        } else {
            (x, n)
        };
        let mut result = 1.0;
        while n > 0 {
            if n & 1 == 1 {
                result *= x;
            }
            x *= x;
            n >>= 1;
        }
        result
    }
}